service/controller/app/v1/resource/status/create.go

```diff
@@ -0,0 +1,54 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	name := key.AppName(cr)
+
+	g8sClient, err := r.kubeConfig.NewG8sClientForApp(ctx, cr)
```

Comment:
That resource is now pretty expensive and takes a rather long time to reconcile constantly. Would it make sense to not compute the client and fetch the chart status in case the status was already set to what we want it to be?

Reply:
Good catch, we don't need to create instance even though it's not needed

Reply:
We can do the same trick we do for BYOC AWS clients. We can store it in the context by one resource and then fetch in other.

Reply:
Sorry to back and forth, but we need this g8sclient to know that the chart status in tenant cluster, otherwise we would never know which status we want to set to `app` status at the first place.

Reply:
Then let's do what Pawel suggested with the context. In case the app status is used in multiple places, we should only fetch it once and put it into the controller context. Then we do not have to fetch it multiple times. The context filling could either be done in a separate resource depending on complexity, or in the controller init function.

Reply:
> We can store it in the context by one resource and then fetch in other. ❤️ 

I like that a lot. We do need to check the tenant cluster for App Catalog but this would reduce the amount of processing needed.

https://github.com/giantswarm/aws-operator/blob/master/service/controller/v22/controllercontext/context.go

@tomahawk28 Here is an example from aws-operator. I'm not deep into the approach used on BYOC (Bring Your Own Cloud) but you can check with Pawel and Theo.

Reply:
I can do add `g8sClient` inside of context, rather than `status` itself, is that right? 
Since we need to check the status regularly, caching it doesn't really help I guess.

Reply:
Yes you add g8sClient to the context.

Reply:
There are few constraints still. 
1. What if `app` CR update their `secret` name? or content itself? 
Then we should invalidate g8sClient inside of running operator, but it's far more complicated then this PR :)

2. Different `app` object meta name and ns
We should have different g8sclient for each of app CR I guess, but is it okay?

Reply:
The client is only added to the context which is about the scope of the reconciliation loop. I think the secret name will not change while a loop is executed. When the secret name changes the operator code changes and thus will be redeployed, causing further loops to be executed in the aligned way. 

Adding the client in general is a nice best practise we should maybe also document for SIG Operator. Also consider doing the same for the status itself in case this is used in multiple places. This would then be another level of optimization.

Reply:
We talked. This is already implemented.